### DISCUSSION TITLE
Use a relationship object attribute in ModelView (how to avoid "AttributeError while querying")

### DISCUSSION THREAD

**Discussion by laurent-leconte** [aminalaee/sqladmin#426] [Q&A]
Hello everyone,

just discovered the sqladmin library, kudos on a great tool!

I wonder if it's possible to access and display a relationship object's _attribute_ in a ModelView?

Consider the following model:
```
class User(Base):

    __tablename__ = "user"
    username = Column(Text())
    email = Column(Text())
    id = Column(UUID(as_uuid=True), primary_key=True)

class AugmentedUser(Base)
     __tablename__ = "augmenteduser"
    phone_number = Column(Text())
    id = Column(UUID(as_uuid=True), primary_key=True)
    user_id = Column(ForeignKey("user.id"), unique=True)
    user = relationship("User", uselist=False, back_populates="augmenteduser")
 ```

I can define a view model for AugmentedUser that links to User:

```
class AugmentedUserAdmin(ModelView, model=AugmentedUser):
    column_list = [AugmentedUser.id, AugmentedUser.phone_number, AugmentedUser.user]
```

However, I want to display the `User.email` attribute, rather than the `User` object.

The following code doesn't work.

```
class AugmentedUserAdmin(ModelView, model=AugmentedUser):
    column_list = [AugmentedUser.id, AugmentedUser.phone_number, AugmentedUser.user.email]
```

This raises `AttributeError: Neither 'InstrumentedAttribute' object nor 'Comparator' object associated with AugmentedUser.user has an attribute 'email'`.

This [Stack Overflow question](https://stackoverflow.com/questions/16589208/attributeerror-while-querying-neither-instrumentedattribute-object-nor-compa) indicates that this is an SQLAlchemy error that can be fixed by accessing the instance (through a join) rather than the class. Is there any way to do this with SQLAdmin?

**Comment by aminalaee** [comment]
Hello and thank you,
No unfortunately this is not possible at the moment.
Though when available you have to do it with a string to avoid the sqlalchemy error like:
column_list = [“user.email”]
